### PR TITLE
Use a mktemp random directory for installation

### LINKED ISSUE #241: Failed to create the file /tmp/helm-diff.tgz: Permission denied
This happens with both helm v2.14.3 and v3.3.0

Steps to reproduce:

    touch /tmp/helm-diff.tgz
    chmod 000 /tmp/helm-diff.tgz
    helm plugin install https://github.com/databus23/helm-diff

Expected results:

    helm downloads the appropriate tarball, unpackages it, puts the contents in their places in $HOME/.helm/
    helm removes the temporary file that was downloaded after using it

Seen instead:

Warning: Failed to create the file /tmp/helm-diff.tgz: Permission denied
curl: (23) Failed writing body (0 != 16360)
Failed to install helm-diff

The "Steps to reproduce" are a contrived scenario, but I encountered it naturally when working on a machine I share with other users. In my case, another user launched helm plugin install weeks before I did.

Workaround:
    sudo rm /tmp/helm-diff.tgz

Subsequent problem:

    tar: diff/LICENSE: Cannot open: File exists
    tar: diff/bin: Cannot utime: Operation not permitted
    tar: diff/plugin.yaml: Cannot open: File exists
    tar: diff/README.md: Cannot open: File exists
    tar: diff/bin/diff: Cannot open: File exists
    tar: diff: Cannot utime: Operation not permitted
    tar: Exiting with failure status due to previous errors
    Failed to install helm-diff

Workaround step two:

    sudo rm -rf /tmp/helm-diff/

### PR DESCRIPTION
**Pull request by keymon**
Do not hardcode a static directory for temporary files, as it
might cause problems when running in a multi-user system
(ie: orphan files owned by other users).

Use mktemp -d instead.

Fixes #241

### CODE REVIEW

**Comment by keymon** [comment]
Pending test on MacOSX

**Comment by zzbennett** [comment]
tried the commands on macos. 

fist command:
```
$ PROJECT_NAME=helm-diff
$ HELM_TMP="$(mktemp -d -t "${PROJECT_NAME}-XXXX")"
$ ls -ld ${HELM_TMP}
$ rmdir ${HELM_TMP}
drwx------  2 lbennett  staff  64 Mar 18 09:53 /var/folders/x4/f_gcn25n0t3ftd25xmb315nh0000gp/T/helm-diff-XXXX.Vx9FxJwQ
```

second command:
```
$ cd /tmp
$ git clone https://github.com/keymon/helm-diff.git helm-diff-keymon
$ cd helm-diff-keymon
$ git checkout plugin-install-mktemp
$ HELM_PLUGIN_DIR=/tmp/helm-plugins ./install-binary.sh
Cloning into 'helm-diff-keymon'...
remote: Enumerating objects: 3, done.
remote: Counting objects: 100% (3/3), done.
remote: Compressing objects: 100% (3/3), done.
remote: Total 816 (delta 0), reused 1 (delta 0), pack-reused 813
Receiving objects: 100% (816/816), 268.29 KiB | 974.00 KiB/s, done.
Resolving deltas: 100% (479/479), done.
Branch 'plugin-install-mktemp' set up to track remote branch 'plugin-install-mktemp' from 'origin'.
Switched to a new branch 'plugin-install-mktemp'
Downloading https://github.com/databus23/helm-diff/releases/download/v3.1.3/helm-diff-macos.tgz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   625  100   625    0     0   2551      0 --:--:-- --:--:-- --:--:--  2551
100 14.7M  100 14.7M    0     0  2395k      0  0:00:06  0:00:06 --:--:-- 2588k
Preparing to install into /tmp/helm-plugins
helm-diff installed into /tmp/helm-plugins/helm-diff

The Helm Diff Plugin

* Shows a diff explaining what a helm upgrade would change:
    This fetches the currently deployed version of a release
  and compares it to a local chart plus values. This can be
  used visualize what changes a helm upgrade will perform.

* Shows a diff explaining what had changed between two revisions:
    This fetches previously deployed versions of a release
  and compares them. This can be used visualize what changes
  were made during revision change.

* Shows a diff explaining what a helm rollback would change:
    This fetches the currently deployed version of a release
  and compares it to adeployed versions of a release, that you
  want to rollback. This can be used visualize what changes a
  helm rollback will perform.

Usage:
  diff [flags]
  diff [command]

Available Commands:
  release     Shows diff between release's manifests
  revision    Shows diff between revision's manifests
  rollback    Show a diff explaining what a helm rollback could perform
  upgrade     Show a diff explaining what a helm upgrade would change.
  version     Show version of the helm diff plugin

Flags:
      --allow-unreleased             enables diffing of releases that are not yet deployed via Helm
  -C, --context int                  output NUM lines of context around changes (default -1)
      --detailed-exitcode            return a non-zero exit code when there are changes
      --devel                        use development versions, too. Equivalent to version '>0.0.0-0'. If --version is set, this is ignored.
      --disable-openapi-validation   disables rendered templates validation against the Kubernetes OpenAPI Schema
      --disable-validation           disables rendered templates validation against the Kubernetes cluster you are currently pointing to. This is the same validation performed on an install
      --dry-run                      disables cluster access and show diff as if it was install. Implies --install, --reset-values, and --disable-validation
  -h, --help                         help for diff
      --include-tests                enable the diffing of the helm test hooks
      --install                      enables diffing of releases that are not yet deployed via Helm (equivalent to --allow-unreleased, added to match "helm upgrade --install" command
      --kubeconfig string            This flag is ignored, to allow passing of this top level flag to helm
      --no-color                     remove colors from the output
      --no-hooks                     disable diffing of hooks
      --output string                Possible values: diff, simple, json, template. When set to "template", use the env var HELM_DIFF_TPL to specify the template. (default "diff")
      --post-renderer string         the path to an executable to be used for post rendering. If it exists in $PATH, the binary will be used, otherwise it will try to look for the executable at the given path
      --reset-values                 reset the values to the ones built into the chart and merge in any new values
      --reuse-values                 reuse the last release's values and merge in any new values. If '--reset-values' is specified, this is ignored
      --set stringArray              set values on the command line (can specify multiple or separate values with commas: key1=val1,key2=val2)
      --set-file stringArray         set values from respective files specified via the command line (can specify multiple or separate values with commas: key1=path1,key2=path2)
      --set-string stringArray       set STRING values on the command line (can specify multiple or separate values with commas: key1=val1,key2=val2)
      --show-secrets                 do not redact secret values in the output
      --suppress stringArray         allows suppression of the values listed in the diff output
  -q, --suppress-secrets             suppress secrets in the output
  -f, --values valueFiles            specify values in a YAML file (can specify multiple) (default [])
      --version string               specify the exact chart version to use. If this is not specified, the latest version is used

Additional help topics:
  diff

Use "diff [command] --help" for more information about a command.
```

File: install-binary.sh [hunk 1]
```diff
@@ -134,6 +143,7 @@ initArch
 initOS
 verifySupported
 getDownloadURL
+mkTempDir
 downloadFile
 installFile
 testVersion
```
Comment (arvindth) [inline install-binary.sh:149]: @keymon don't you want to `rmTempDir` here as well in case of successful run?

Comment (arvindth) [reply install-binary.sh:149]: Ah, it traps on all shell exits. Never mind.

**Pull request closed without merging**